src/lexer.rs

```diff
@@ -125,6 +125,9 @@ impl<'a> Iterator for Lexer<'a> {
 					Some(_) => self.characters.next(),
 				};
 			},
+			'[' => Token::SquareOpen,
+			']' => Token::SquareClose,
+			'_' => Token::Underscore,
```

Comment:
Normally, I'd put these at the top of the match statement where all the other singular tokens are but this is fine too.

Reply:
Done